flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/CommonTestUtils.java

```diff
@@ -203,10 +203,12 @@ public static void waitUtil(Supplier<Boolean> condition, Duration timeout, Strin
             throw new IllegalArgumentException("The timeout must be positive.");
         }
         long startingTime = System.currentTimeMillis();
-        while (!condition.get() && System.currentTimeMillis() - startingTime < timeoutMs) {
+        boolean conditionResult = false;
```

Comment:
```suggestion
        boolean conditionResult = condition.get();
```
Is it necessary to do one loop here?

Reply:
This also seems like a bugfix commit :)